app/kubernetes/components/datatables/applications-datatable/applicationsDatatableController.js

```diff
@@ -25,13 +77,29 @@ angular.module('portainer.docker').controller('KubernetesApplicationsDatatableCo
     };
 
     this.isSystemNamespace = function (item) {
+      // if all charts in a helm app/release are in the system namespace
+      if (item.KubernetesApplications && item.KubernetesApplications.length > 0) {
+        return item.KubernetesApplications.some((app) => KubernetesNamespaceHelper.isSystemNamespace(app.ResourcePool));
+      }
       return KubernetesNamespaceHelper.isSystemNamespace(item.ResourcePool);
     };
 
     this.isDisplayed = function (item) {
       return !ctrl.isSystemNamespace(item) || ctrl.settings.showSystem;
     };
 
+    this.getPublishUrl = function (item) {
+      // Map all ingress rules in published ports to their respective URLs
+      const publishUrls = item.PublishedPorts.flatMap((pp) => pp.IngressRules).map(({ Host, IP, Path }) => `http://${Host || IP}${Path}`);
+
+      // Return the first URL
+      return publishUrls.length > 0 ? publishUrls[0] : '';
+    };
+
+    this.hasConfigurationSecrets = function (item) {
```

Comment:
(from slack) you're basically doing the secret extraction twice. once to check if there are secrets and once in the component. so why not make this function return the secrets. (just changed .some to .map), and make the component get an array of secrets